### PR TITLE
Add lock path when returning ErrAlreadyLocked

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

When returning the error `data path already locked by another beat`, add the path to the lock file to the error message

## Why is it important?

This error is likely due a miss configuration due to a user change, or some inconsistent state. Either way, without deep knowledge about the beats internal it isn't trivial to determine "what" is locking the beat, and where the lock is. Hopefully, by adding the lock file path to the error message, a regular user, not a beats developer, will be able to fully understand the error, and eventually fix it.

## Checklist

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Ensure a lock file is present when starting the beat. 
It can be done by running 2 beats sharing the same `data.path`, or just stopping the beat or elastic-agent, if run under the Elastic Agent, creating the lock file, and starting the beat/elastic-agent again.

for a regular filebeat, the lock file is on `Elastic/Agent/data/elastic-agent-*/run/default/filebeat--8.x.y/filebeat.lock`

## Use cases

Investigating why a beat does not start or an Elastic Agent is unhealthy due to `Exiting: data path already locked by another beat. Please make sure that multiple beats are not sharing the same data path (path.data).`

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -11,7 +11,7 @@ https://github.com/elastic/beats/compare/v8.2.0\...main[Check the HEAD diff]
 *Affecting all Beats*
 
 * Upgrade to Go 1.18. Certificates signed with SHA-1 are now rejected. See the Go 1.18 https://tip.golang.org/doc/go1.18#sha1[release notes] for details. {pull}32493[32493]
-
+* Add lock path when returning ErrAlreadyLocked. {pull}[32980]
```
Comment (belimawr) [inline CHANGELOG.next.asciidoc:14]: I believe it does not make sense to mention `ErrAlreadyLocked` on this changelog. If it was the `CHANGELOG-developer.next.asciidoc` it would be ok.

Maybe rephrase to something like:
> Add lock path to error message when the lock file already exists.

Comment (belimawr) [reply CHANGELOG.next.asciidoc:14]: Also the format for the link at the end should be: `{pull}32980[32980]`

**Review by belimawr — APPROVED** [review]

**Review by michalpristas — APPROVED** [review]
looks good

**Comment by jlind23** [comment]
Closing for now.
